### PR TITLE
Vended items now go into hand

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Items that you buy or vend from a vending machine will now be transferred straight into your hand, provided certain conditions are met. 

## Why It's Good For The Game

Prevents people from yoinking your items and is generally a nice thing to have. QOL.

## Changelog
:cl:
tweak: Vended items now go straight into your hand.
/:cl:

### CODE REVIEW

**Comment by justinbooj9** [comment]
just by seeing the PR name I already love it

**Comment by Krysonism** [comment]
I like items popping up over the vendor sprite.

**Comment by MrDoomBringer** [comment]
>Prevents people from yoinking your items

We want conflict to be easy to create among players

**Comment by Gandalf2k15** [comment]
> > Prevents people from yoinking your items
> 
> We want conflict to be easy to create among players

I don't really think it should be that easy to just steal someone's item. And it's much more a QOL change,

**Comment by Jared-Fogle** [comment]
but its funny 🙁

**Comment by Yenwodyah** [comment]
you can still yoink with this, you just have to get behind the person and shove them.

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -866,8 +866,14 @@ GLOBAL_LIST_EMPTY(vending_products)
 			if(icon_vend) //Show the vending animation if needed
 				flick(icon_vend,src)
 			playsound(src, 'sound/machines/machine_vend.ogg', 50, TRUE, extrarange = -3)
-			new R.product_path(get_turf(src))
+			var/vended = new R.product_path(get_turf(src))
 			R.amount--
+			if(istype(vended, /obj/item))
```
Comment (Rohesie) [inline code/modules/vending/_vending.dm:871]: Can anything that's not an item be vended?

Comment (Gandalf2k15) [reply code/modules/vending/_vending.dm:871]: I am not sure, but this prevents any edge cases which would cause runtimes. Better safe than sorry.

**Comment by cacogen** [comment]
I don't even know if TK is still in the game but at some prior point this would create issues with TK as it would allow people to teleport items into their hands

**Comment by KathrinBailey** [comment]
Now you gotta actually fight someone for the insuls they just bought to steal them instead of ninja-grabbing them by standing next to the vendor and just spam clicking.

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/vending/_vending.dm:871]: Silent sanity checks are bad in the long term, as if something does fail we'll have a hard time tracking it.
Just remove the check entirely, we want errors situations to runtime.
And if you must add a sanity check anywhere, do so with an informative `stack_trace` or `CRASH`, as relevant.

**Comment by TacticalSpacePilot** [comment]
if my hands are full will the vendor still vend like before

**Comment by cacogen** [comment]
If you aren't adjacent (e.g. TK) or it can't be put into your hands it falls onto the floor instead.

**Comment by Rohesie** [comment]
> 
> 
> If you aren't adjacent (e.g. TK) or it can't be put into your hands it falls onto the floor instead.

Falling on the floor means the old behavior of spawning in the tile of the vendor, though.

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/modules/vending/_vending.dm]: Agreed, I've updated it.

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -866,8 +866,12 @@ GLOBAL_LIST_EMPTY(vending_products)
 			if(icon_vend) //Show the vending animation if needed
 				flick(icon_vend,src)
 			playsound(src, 'sound/machines/machine_vend.ogg', 50, TRUE, extrarange = -3)
-			new R.product_path(get_turf(src))
+			var/obj/item/V = new R.product_path(get_turf(src))
```
Comment (Rohesie) [inline code/modules/vending/_vending.dm:869]: You used a nice `vended` variable, why do you have to downgrade it to a single capital letter? `vended_item` would be even better

Comment (Gandalf2k15) [reply code/modules/vending/_vending.dm]: Done!

**Comment by Rohesie** [comment]
Is this something you want to block, @Jared-Fogle ? Else I'm inclined to try it.

**Comment by Timberpoes** [comment]
Memes aside, I'd personally consider this closer to a very minor QoL change. You can still shove people from behind to steal their vending machine gamer gear.

Would love to see it merged, but would not lose any sleep if it was blocked.

**Comment by Jared-Fogle** [comment]
I'm not blocking this, no.

**Pull request merged**